vllm/attention/backends/rocm_flash_attn.py

```diff
@@ -509,16 +518,18 @@ def __init__(
                                "FA backend instead by setting the env var "
                                "`VLLM_USE_TRITON_FLASH_ATTN=0`")
         else:
-            # if not using triton, navi3x/navi21/navi10 do not use flash-attn
+            # if not using triton, navi21/navi10 do not use flash-attn
             # either
             if not current_platform.has_device_capability(90):
                 self.use_naive_attn = True
             else:
-                try:
-                    from flash_attn import flash_attn_varlen_func  # noqa: F401
-                    self.attn_func = flash_attn_varlen_func
+                if flash_attn_available:
+                    if current_platform.has_device_capability(110):
```

Comment:
This check isn't equivalent to is_navi
On Cuda device_capability is meant to increase with each new architecture, and is meant to differentiate by new features support, such as FP8, etc.
On ROCm it is 1st digit of gfx * 10 + 2nd digit, which doesn't mean much, especially for any future architectures.

Reply:
updated